src/scripts/btma/annotator.py

```diff
@@ -63,81 +65,101 @@ def main():
         mod = omn_path / "{m}.omn".format(m=module)
         exp = csv_path / "{m}.csv".format(m=module)
         sp.call('java -jar {jar} export --input {m} \
-                                    --header "{c}" \
-                                    --export {e}'.format(jar=robot_path,
-                                                           m=mod, 
-                                                           c=col, 
-                                                           e=exp))
-        tmp = extract_IDs(module)
+                                        --header "{c}" \
+                                        --export {e}'.format(jar=robot_path,
+                                                             m=mod, 
+                                                             c=col, 
+                                                             e=exp))
+        tmp = extract_IDs(csv_path / (module + '.csv'))
         ids = ids.union(set(tmp))
         ids_per_module[i] = tmp
 
     ids = list(ids)
 
-    # Create overview about IRIs and there belonging
+    # Compare new data with database
 
-    print('Finding belongings...')
+    old_ids_per_module = [_,_,_,_]
+    try:
+        for i, module in enumerate(modules):
+            old_ids_per_module[i] = extract_IDs(db_path / (module + '.csv'))
+    except:
+        pass
+        
+    if old_ids_per_module != ids_per_module:
 
-    belonging = ids.copy()
+        # Create overview about IRIs and there belonging
 
-    for (i,id) in enumerate(ids):
-        for (j,module_ids) in enumerate(ids_per_module):
-            if id in module_ids:
-                belonging[i] = modules[j]
-                break
+        print('Finding belongings...')
 
-    # Create the CSV template with the help of the overview
+        belonging = ids.copy()
 
-    print('Creating template...')
+        for (i,id) in enumerate(ids):
+            for (j,module_ids) in enumerate(ids_per_module):
```

Comment:
You are getting duplicates of the annotations, check if there are duplicates here before you add them 
![Screenshot 2022-04-14 at 12 52 51](https://user-images.githubusercontent.com/512096/163377204-d6dc9eaa-7893-4ebc-a34a-b4eebd2fca80.png)

Reply:
There now seem to be no problems with duplicate annotations due to previous changes.(I had run some automated checks, too.)
But now there is the problem that entities that do not belong to the module at all are annotated to their associated modules here.

This breaks the modularization for Protege.
I try to solve it now.
<img width="551" alt="image" src="https://user-images.githubusercontent.com/86624154/163714834-4418ac6b-c54b-4154-9cbc-fadeb501fb21.png">

Reply:
Weird.
I've run the script againg and reloaded protege.
Now it looks fine.
I don't know why.
But as long as it works, I don't think it matters.<img width="560" alt="image" src="https://user-images.githubusercontent.com/86624154/163715387-1158b30d-1f7e-4454-8a48-035c9b5e6db9.png">